specs-go/state.go

```diff
@@ -33,3 +35,17 @@ type State struct {
 	// Annotations are key values associated with the container.
 	Annotations map[string]string `json:"annotations,omitempty"`
 }
+
+type SeccompState struct {
+	// Version is the version of the specification that is supported.
+	Version string `json:"ociVersion"`
+	// SeccompFd is the file descriptor for Seccomp User Notification
+	SeccompFd int `json:"seccompFd"`
```

Comment:
How does it work when multiple hooks are specified?

Reply:
This is not something that I would encourage users to do: I can't see a real use case where that would be an useful thing to do.

But from the container runtime perspective, that's fine: all hooks are called one after another and each of them would receive the same seccomp fd. If there are several seccomp agents that have the same seccomp fd but only one of them does something with it, that would be fine. If they concurrently run `ioctl(SECCOMP_IOCTL_NOTIF_RECV)`, I am not sure what would happen.

Reply:
It seems better to explicitly prohibit registering multiple hooks.

Reply:
Sounds fine to me. Would you prohibit it in the spec text or also change the type from an array to a single element?

```diff
-SendSeccompFd []Hook
+SendSeccompFd *Hook
```

Reply:
I'd prefer the latter one

Reply:
could it be a more generic implementation, something like adding a new hook type: "PreStartRuntime" that is called in the runtime namespace when the container is ready to start?

It could accept a set of "name" -> "fd" so that it can be extended in future if needed.  We could use it for the tty as well.